pkg/providers/instancetype/instancetype.go

```diff
@@ -101,27 +101,26 @@ func (p *DefaultProvider) List(ctx context.Context, kc *corev1beta1.KubeletConfi
 	defer p.muInstanceTypeInfo.RUnlock()
 	defer p.muInstanceTypeOfferings.RUnlock()
 
+	if kc == nil {
+		kc = &corev1beta1.KubeletConfiguration{}
+	}
+	if nodeClass == nil {
```

Comment:
musing: Is there a way that we can remove this nil check in this List call long-term

Reply:
We will need to make sure that users don't pass in nil values, but I can see us getting rid of this in the future

Reply:
I'm wondering if we can just track down the calls here and determine that we will never get a `nil` call here. I think that I originally built it this way so that we can still resolve InstanceTypes with default values, even if we don't get any offerings, but it may not be worth this configuration or handling anymore.

Anyways, we can punt this to a separate change and come back to it later.